file-formats/aifi/type/zif_aff_aifi_v1.intf.abap

```diff
@@ -0,0 +1,506 @@
+      "! <p class="shorttext">Selection Type</p>
+      "! Selection type
+      " (type /aif/multi_sel_type)
+      selection_type           TYPE ty_selection_type,
+      "! <p class="shorttext">Message Index Table</p>
+      "! Message index table
+      " (type /aif/msg_tbl)
+      idx_table                TYPE zif_aff_types_v1=>ty_object_name_30,
+      "! <p class="shorttext">Parent Field Sequence Number</p>
+      "! Parent field sequence number
+      " (type /aif/field_number_parent)
+      parent_field_sequence    TYPE n LENGTH 3,
+      "! <p class="shorttext">Icon</p>
+      "! Icon
+      " (type /aif/icon_l4)
+      icon                     TYPE c LENGTH 4,
+      "! <p class="shorttext">Icon Tooltip</p>
+      "! Icon tooltip
+      " (type /aif/tooltip)
+      icon_tooltip             TYPE c LENGTH 40,
+      "! <p class="shorttext">Alert Recipient Field Name</p>
+      "! Field name in alert recipient assignment table
+      " (type /aif/alertfieldname)
+      alert_field_name         TYPE c LENGTH 45,
+      "! <p class="shorttext">Category Field Name</p>
+      "! Category field name
+      " (type /aif/alrt_catfn)
+      cat_fieldname            TYPE c LENGTH 32,
+      "! <p class="shorttext">Show Tree Node</p>
+      "! Show tree node in view 1 tree
+      "! $default 'X'
+      " /AIF/IS_HIDDEN, swap to positive stature
+      show_tree_node           TYPE abap_bool,
+      "! <p class="shorttext">Relevant for Alert Recipient Determination</p>
+      "! Relevant for alert recipient determination
+      " (type /aif/alertrelevant)
+      alert_recipient_relevant TYPE abap_bool,
+    END OF ty_keyfield_selection.
+
+  TYPES:
+    "! <p class="shorttext">SAP Fiori Features</p>
+    "! SAP Fiori specific features
+    BEGIN OF ty_fiori_features,
+      "! <p class="shorttext">Semantic Object</p>
+      "! Semantic object
+      " (type /aif/semantic_object)
+      semantic_object         TYPE zif_aff_types_v1=>ty_object_name_30,
+      "! <p class="shorttext">Semantic Action</p>
+      "! Semantic action
+      " (type /aif/semantic_action)
+      semantic_action         TYPE zif_aff_types_v1=>ty_description_60,
+      "! <p class="shorttext">Display in Message Monitoring</p>
+      "! Display in message monitoring
+      "! $default 'X'
+      " /AIF/NOT_SHOW_FIELD
+      message_monitor_show    TYPE abap_bool,
+      "! <p class="shorttext">Display in Message Monitoring by Default</p>
+      "! Display in message monitoring by default
+      " /AIF/IS_DEFAULT_FIELD
+      message_monitor_default TYPE abap_bool,
+    END OF ty_fiori_features.
+
+  TYPES:
+    "! <p class="shorttext">Key Field Rule Fields</p>
+    "! Key field rule fields
+    " Related key fields used by key field rules
+    BEGIN OF ty_key_field_rules_field,
+      "! <p class="shorttext">Key Field Name</p>
+      "! Key field name
+      " (type /aif/key_fieldname_enh)
+      key_field_rule_field TYPE zif_aff_types_v1=>ty_object_name_30,
```

Comment:
The name is a bit confusing with two `fields`. Do you have a better idea? Maybe `key_field_name` like in the title?

Reply:
I agree, but it´s kind of tricky. It´s the name of a key field used in a key field rule. I propose 'rule_key_field_name'